### PR TITLE
[PRTL-2567] improve variable grouping and sorting

### PR DESCRIPTION
**Pull request by caravinci**

### CODE REVIEW

File: src/packages/@ncigdc/containers/explore/presetFacets.ts [hunk 1]
```diff
@@ -116,14 +116,157 @@ export const clinicalFacets: IClinicalFacetProps[] = [
...
+    'molecular_consequence',
+    'ploidy',
+    'second_gene_symbol',
+    'test_analyte_type',
+    'variant_type',
+    'zygosity',
+  ],
+}
+
+export const customSorting = (field: string = ''): object | number => {
```
Comment (caravinci) [inline src/packages/@ncigdc/containers/explore/presetFacets.ts]: This is where the magic happens. Pretty much every other modified file is just formatting.

File: src/packages/@ncigdc/containers/explore/presetFacets.ts [hunk 2]
```diff
@@ -116,14 +116,157 @@ export const clinicalFacets: IClinicalFacetProps[] = [
...
-  'diagnoses.ajcc_clinical_t': -23,
-  'diagnoses.ajcc_clinical_n': -22,
-  'diagnoses.ajcc_clinical_m': -21,
-  'diagnoses.ajcc_clinical_stage': -20,
-  'diagnoses.ajcc_pathologic_t': -13,
-  'diagnoses.ajcc_pathologic_n': -12,
-  'diagnoses.ajcc_pathologic_m': -11,
-  'diagnoses.ajcc_pathologic_stage': -10,
-  'diagnoses.ajcc_staging_system_edition': -1,
+const clinicalFacetSortingDictionary = {
```
Comment (caravinci) [inline src/packages/@ncigdc/containers/explore/presetFacets.ts]: This dictionary will allow for customised sorting without having to change the numbers manually, unlike i had designed it initially. Just add a new value exactly where you want it, and Bob's your uncle 👍

File: src/packages/@ncigdc/containers/explore/ClinicalAggregations.tsx [hunk 3]
```diff
@@ -526,7 +526,7 @@ const enhance = compose(
                                       ),
                                     )}
                                     style={{
-                                      order: customSorting[componentFacet.field] || 0,
+                                      order: customSorting(componentFacet.field) || 0,
```
Comment (caravinci) [inline src/packages/@ncigdc/containers/explore/ClinicalAggregations.tsx]: This calls the sorting order calculator, rather than just asking a dictionary.

Comment (yuanhengzhao) [reply src/packages/@ncigdc/containers/explore/ClinicalAggregations.tsx]: The order of all facets here is different from the order in line 461. Should change the asc in line 464.

Comment (yuanhengzhao) [reply src/packages/@ncigdc/containers/explore/ClinicalAggregations.tsx]:
```suggestion
                                      order: customSorting(componentFacet.field),
```

**Pull request merged**